Catch "No space left on device" from g++

We could catch if a "No space left on device" text appears in stderr when building cmdstan or compiling a model and make a warning that is more easily readable. These C++ compiler error messages can get lost in the wall of text they produce.

Not super urgent but a quality of life improvement nonetheless.

Example of output: 

```
g++ -o task_group_context.o -c -MMD -DDO_ITT_NOTIFY -O2 -DUSE_PTHREAD -m64 -mrtm  -fPIC  -D__TBB_BUILD=1 -Wall -Wextra -Wno-parentheses -Wno-sized-deallocation -Wno-unknown-warning-option -Wno-missing-attributes -Wno-class-memaccess -Wno-sized-deallocation   -std=c++1y  -I../tbb_2019_U8/src -I../tbb_2019_U8/src/rml/include -I../tbb_2019_U8/include ../tbb_2019_U8/src/tbb/task_group_context.cpp
../tbb_2019_U8/src/tbb/task_group_context.cpp:491:1: fatal error: error writing to /tmp/ccaVlne4.s: No space left on device
```

Need to check what clang produces but should be something close to this.

--- a/R/model.R
+++ b/R/model.R
@@ -513,6 +513,13 @@ compile <- function(quiet = TRUE,
           call. = FALSE
         )
       }
+      if (grepl("No space left on device", x) || grepl("error in backend: IO failure on output stream", x)) {
+        warning(
+          "The C++ compiler ran out of disk space and was unable to build the executables for your model!\n",
+          "See the above error for more details.",
+          call. = FALSE
+        )
+      }
     },
     error_on_status = FALSE
   )

